consensus/hotstuff/votecollector/statemachine.go

```diff
@@ -11,23 +11,53 @@ import (
 	"github.com/onflow/flow-go/consensus/hotstuff"
 	"github.com/onflow/flow-go/consensus/hotstuff/model"
 	"github.com/onflow/flow-go/consensus/hotstuff/voteaggregator"
+	"github.com/onflow/flow-go/module/irrecoverable"
+	"github.com/onflow/flow-go/utils/logging"
 )
 
 var (
 	ErrDifferentCollectorState = errors.New("different state")
 )
 
-// VerifyingVoteProcessorFactory generates hotstuff.VerifyingVoteCollector instances
+// VerifyingVoteProcessorFactory generates [hotstuff.VerifyingVoteProcessor] instances
 type VerifyingVoteProcessorFactory = func(log zerolog.Logger, proposal *model.SignedProposal) (hotstuff.VerifyingVoteProcessor, error)
 
-// VoteCollector implements a state machine for transition between different states of vote collector
+// VoteCollector implements a state machine for transition between different states of vote processor.
+// It ingests *all* votes for a specified view and consolidates the handling of all byzantine edge cases
+// related to voting (including byzantine leaders publishing conflicting proposals and/or votes).
+// On the happy path, the VoteCollector generates a QC when enough votes have been ingested.
+// We internally delegate the vote-format specific processing to the VoteProcessor.
 type VoteCollector struct {
 	sync.Mutex
```

Comment:
I am not sure, but think we should be able to remove this mutex by now. The mutex is only used in the functions [`caching2Verifying` and `terminateVoteProcessing`](https://github.com/onflow/flow-go/blob/946192e08854f65389c2d7ececf3e10148657e36/consensus/hotstuff/votecollector/statemachine.go#L363-L392). In both cases, I think we can use compare and swap operations on the [atomic `votesProcessor`](https://github.com/onflow/flow-go/blob/946192e08854f65389c2d7ececf3e10148657e36/consensus/hotstuff/votecollector/statemachine.go#L62). This is optional but would be a nice clean up and improvement of design consistency.